tests/vmi_multus_test.go

```diff
@@ -871,14 +879,76 @@ var _ = Describe("SRIOV", func() {
 	})
 })
 
+var _ = Describe("Macvtap", func() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+	macvtapNetworkName := "net1"
+
+	createCirrosVMIWithMacvtapStaticIP := func(virtClient kubecli.KubevirtClient, networkName string, ifaceName string, ipCIDR string, mac *string) *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithMacvtapInterfaceEphemeralDiskAndUserdata(tests.ContainerDiskFor(tests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n", networkName)
+		if mac != nil {
+			vmi.Spec.Domain.Devices.Interfaces[0].MacAddress = *mac
+		}
+		_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).ToNot(HaveOccurred())
+
+		// configure the client VMI
+		tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+		configInterface(vmi, ifaceName, ipCIDR, "\\$ ", true)
+		return vmi
+	}
+
+	tests.BeforeAll(func() {
+		// provision macvtap NAD
+		result := virtClient.RestClient().
+			Post().
+			RequestURI(fmt.Sprintf(postUrl, tests.NamespaceTestDefault, macvtapNetworkName)).
+			Body([]byte(fmt.Sprintf(macvtapNetworkConf, macvtapNetworkName, tests.NamespaceTestDefault, macvtapNetworkName))).
+			Do()
+		Expect(result.Error()).NotTo(HaveOccurred())
+	})
+
+	Context("a virtual machine with one macvtap interface, with a custom MAC address", func() {
+		var serverVMI *v1.VirtualMachineInstance
+		chosenMAC := "de:ad:00:00:be:af"
+		serverCIDR := "10.10.10.102/24"
+
+		BeforeEach(func() {
+			serverVMI = createCirrosVMIWithMacvtapStaticIP(virtClient, macvtapNetworkName, "eth0", serverCIDR, &chosenMAC)
+		})
+
+		It("should have the specified MAC address reported back via the API", func() {
+			vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(serverVMI.Name, &metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(vmi.Status.Interfaces)).To(Equal(1))
+			Expect(vmi.Status.Interfaces[0].MAC).To(Equal(chosenMAC))
+		})
+
+		Context("another virtual machine is created, then connected to the same macvtap multus network", func() {
```

Comment:
"and another virtual machine connected to the same network" would read better as the whole test name